rustls/src/conn/unbuffered.rs

```diff
@@ -0,0 +1,449 @@
+//! Unbuffered connection API
+
+use alloc::vec::Vec;
+use core::num::NonZeroUsize;
+use core::{fmt, mem};
+use std::error::Error as StdError;
+
+use super::UnbufferedConnectionCommon;
+use crate::msgs::deframer::DeframerSliceBuffer;
+use crate::Error;
+
+impl<Data> UnbufferedConnectionCommon<Data> {
+    /// Processes the TLS records in `incoming_tls` buffer
```

Comment:
I think this doc string could be expanded. The way it's written now makes it sound like it would process all of the records in `incoming_tls` buf, but the impl breaks after specific records/states.